src/views/AuthenticationWrapper/DevSecurity.tsx

```diff
@@ -43,45 +48,52 @@ const DevSecurity = ({ children }: ParentComponentProps) => {
   useEffect(() => {
     if (window.localStorage[storageKey]) {
       const state = JSON.parse(window.localStorage[storageKey]);
-      setAuthState(as => {
+      setAuthState(prevAuthState => {
         return {
-          ...as,
-          name: `User ${state.eua}`,
-          isAuthenticated: true,
-          euaId: state.eua,
-          groups: state.jobCodes
+          ...prevAuthState,
+          isAuthenticated: true
         };
       });
+      setEuaId(state.eua);
+      setGroups(state.jobCodes);
     }
   }, []);
 
-  const handleSubmit: ReactEventHandler = event => {
+  const handleSubmit = (event: React.FormEvent) => {
     event.preventDefault();
+    const activeGroups = jobCodes.filter(group => groups[group]);
     const value = {
       eua: euaId,
-      jobCodes: Array.from(checkboxValues.current)
+      jobCodes: activeGroups
     };
     localStorage.setItem(storageKey, JSON.stringify(value));
     localStorage.removeItem('okta-token-storage'); // ensure that the dev token is used
     setAuthState({
       ...authState,
-      isAuthenticated: true,
-      name: `User ${euaId}`,
-      euaId,
-      groups: Array.from(checkboxValues.current)
+      isAuthenticated: true
     });
   };
 
-  const checkboxChange: ReactEventHandler<HTMLInputElement> = event => {
-    if (event.currentTarget.checked) {
-      checkboxValues.current.add(event.currentTarget.value);
-    } else {
-      checkboxValues.current.delete(event.currentTarget.value);
-    }
+  const checkboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.persist();
+    setGroups(prevGroups => {
+      return {
+        ...prevGroups,
+        [event.target.value]: event.target.checked
+      };
+    });
+  };
+
+  const mockOktaAuthState = {
```

Comment:
Instead of having `authState` be 1:1 to Okta, we can construct an object and pass it through the context.